### PR TITLE
doc/README.md - improve "CMake Options"

### PR DESCRIPTION
**Pull request by zdover23**
Add subheadings to the "CMake Options" section of ceph/README.md. Remove modal verbs when appropriate and make sentences tidier.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,42 +111,53 @@ To install:
  
 ### CMake Options
 
-If you run the `cmake` command by hand, there are many options you can
-set with "-D". For example, the option to build the RADOS Gateway is
-defaulted to ON. To build without the RADOS Gateway:
+The `-D` flag can be used with the `cmake` command to speed up the process of
```
Comment (anthonyeleven) [inline README.md:114]: I would write "can be used with `cmake` to"

File: README.md [hunk 2]
```diff
@@ -111,42 +111,53 @@ To install:
...
 
 	cmake -DCMAKE_INSTALL_PREFIX=/opt/ceph -DCMAKE_C_FLAGS="-Og -g3 -gdwarf-4" \
 	..
 
 Ceph has several bundled dependencies such as Boost, RocksDB and Arrow. By
-default, cmake will build these bundled dependencies from source instead of
-using libraries that are already installed on the system. You can opt-in to
-using these system libraries, provided they meet the minimum version required
-by Ceph, with cmake options like `WITH_SYSTEM_BOOST`:
+default, cmake builds these bundled dependencies from source instead of using
```
Comment (anthonyeleven) [inline README.md:133]: Above we backtick `cmake`

File: README.md [hunk 3]
```diff
@@ -111,42 +111,53 @@ To install:
...
+libraries that are already installed on the system. You can opt to use these
+system libraries, as long as they meet the minimum version required by Ceph. To
```
Comment (anthonyeleven) [inline README.md:135]: s/minimum version required by Ceph/Ceph's version requirements/

File: README.md [hunk 4]
```diff
@@ -111,42 +111,53 @@ To install:
...
+use system libraries, use cmake options like `WITH_SYSTEM_BOOST`, as in the
```
Comment (anthonyeleven) [inline README.md:136]: `cmake` above.

**Review by anthonyeleven — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (zdover23) [reply README.md:133]: I was on the fence about it, but if your fresh eyes expect it, then we do it.

File: README.md (hunk 1 above)
Comment (zdover23) [reply README.md:114]: I like that better.

File: README.md (hunk 3 above)
Comment (zdover23) [reply README.md:135]: Accepted.

File: README.md (hunk 4 above)
Comment (zdover23) [reply README.md:136]: Accepted.

File: README.md (hunk 2 above)
Comment (anthonyeleven) [reply README.md]: Either way, be consistent.

**Pull request merged**